addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusSymoInverterHandler.java

```diff
@@ -1,230 +1,246 @@
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errorMsg);
-        }
-
-        return resultOk ? result : null;
-    }
-
-}
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fronius.handler;
+
+import java.io.IOException;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.fronius.FroniusBaseDeviceConfiguration;
+import org.openhab.binding.fronius.FroniusBindingConstants;
+import org.openhab.binding.fronius.FroniusBridgeConfiguration;
+import org.openhab.binding.fronius.internal.api.InverterRealtimeResponse;
+import org.openhab.binding.fronius.internal.api.PowerFlowRealtimeResponse;
+import org.openhab.binding.fronius.internal.api.ValueUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link FroniusSymoInverterHandler} is responsible for updating the data, which are
+ * sent to one of the channels.
+ *
+ * @author Thomas Rokohl - Initial contribution
+ */
+public class FroniusSymoInverterHandler extends FroniusBaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FroniusSymoInverterHandler.class);
+    private InverterRealtimeResponse inverterRealtimeResponse;
+    private PowerFlowRealtimeResponse powerFlowResponse;
+    private FroniusBaseDeviceConfiguration config;
+    private static final int API_TIMEOUT = 5000;
+    private Gson gson;
+
+    public FroniusSymoInverterHandler(Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    @Override
+    protected String getDescription() {
+        return "Fronius Symo Inverter";
+    }
+
+    @Override
+    public void refresh(FroniusBridgeConfiguration bridgeConfiguration) {
+        updateData(bridgeConfiguration, config);
+        updateChannels();
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(FroniusBaseDeviceConfiguration.class);
+        super.initialize();
+    }
+
+    /**
+     * Update the channel from the last data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     * @return the last retrieved data
+     */
+    @Override
+    protected Object getValue(String channelId) {
+        String[] fields = StringUtils.split(channelId, "#");
+
+        String fieldName = fields[0];
+
+        if (inverterRealtimeResponse == null || inverterRealtimeResponse.getBody() == null

```

Comment:
Here a status check should be done too. The null checks (except for the first) will never be true with the new changes.